python/ray/data/tests/test_autoscaling_coordinator.py

```diff
@@ -410,6 +412,70 @@ def test_coordinator_accepts_zero_resource_for_missing_resource_type(
     assert coordinator.get_allocated_resources("spam") == [{"CPU": 1, "GPU": 0}]
 
 
+def test_double_allocation_with_multiple_request_remaining():
+    """Test fair allocation when multiple requesters have request_remaining=True."""
+    cluster_nodes = [
+        {
+            "Resources": {
+                "CPU": 10,
+                "GPU": 5,
+                "object_store_memory": 1000,
+            },
+            "Alive": True,
+        }
+    ]
+
+    with (
+        patch("ray.nodes", return_value=cluster_nodes),
+        patch("time.time", mock_time),
+        patch("ray.autoscaler.sdk.request_resources"),
+    ):
```

Comment:
Out-of-scope for this PR, but I think this `DefaultAutoscalingCoordinator` would be more maintainable long-term if we used explicit seams rather than implicit depedencies:
```python
class DefaultAutoscalingCoordinator:
    def __init__(
        self,
        ...,
        get_node_resources: Callable[[], List[ResourceDict]],
        get_time: Callable[[], float] = time.time,
        request_resources: Callable[[List[ResourceDict], None] = ray.autoscaler.sdk.request_resources
)
```

The problem with patching is that this test will break if we change some of these implementation details (e.g., use `time.perf_counter` instead of `time.time`), and it's also less clear what the dependencies of the component actually are.

Reply:
Yeah, I think that's totally fine as follow-up. 

> It seems like we need to do for both DefaultAutoscalingCoordinator and _AutoscalingCoordinatorActor?

I looked through the tests, and I think it's okay if we just add it to `_AutoscalingCoordinatorActor` for now. Seems like most of the core unit tests (`test_basic` and the newly-added `test_double_allocation_with_multiple_request_remaining`) are against the `_AutoscalingCoordinatorActor` layer of abstraction, rather than `DefaultAutoscalingCoordinator`